src/squishy/core.clj

```diff
@@ -12,22 +14,46 @@
   (let [fail-message (pr-str {:body body :error (.getMessage error)})]
     (sqs/send client fail-queue-url fail-message)))
 
-(defn safe-process [client fail-queue-url f]
+(defn safe-process
+  [client queue-url fail-queue-url visibility-timeout f]
   (fn [message]
     (log/debug "Processing SQS message:" (str "<<" message ">>"))
-    (try (f message)
-      (catch Exception e
-        (let [body (:body message)]
-          (log/error "Failed to process" body e)
-          (report-error client fail-queue-url body e))))))
+    (let [timeout-increaser (future
+                              (loop [timeout visibility-timeout]
+                                (let [new-timeout (int (* timeout 1.5))]
+                                  (Thread/sleep (* 1000 (- timeout (/ timeout 4))))
```

Comment:
This is just the slightest bit opaque. You want to increase the timeout after 75% of the current one? `(* 1000 timeout 0.75)`